### PR TITLE
Feature/some refactoring

### PR DESCRIPTION
**Pull request by guusvw**

### CODE REVIEW

**Comment by mrIncompetent** [comment]
@guusvw rebase please

Has this been tested?

**Comment by guusvw** [comment]
(Rebase )Will do it and I will also write an test for `EnsureSSHKeypairSecret` to ensure the same behaviour as before.

**Comment by mrIncompetent** [comment]
We indeed are missing some tests

File: pkg/client/clientset/versioned/scheme/register.go [hunk 1]
```diff
@@ -24,13 +24,15 @@ import (
...
+)
 
 func init() {
-	v1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
-	AddToScheme(Scheme)
+	v1.AddToGroupVersion(scheme, schema.GroupVersion{Version: "v1"})
+	AddToScheme(scheme)
 }
 
 // AddToScheme adds all types of this clientset into the given scheme. This allows composition
```
Comment (mrIncompetent) [inline pkg/client/clientset/versioned/scheme/register.go]: everything under pkg/client is generated code

File: pkg/ssh/configmap.go [hunk 2]
```diff
@@ -20,48 +20,59 @@ const (
 	secretName = "machine-controller-ssh-key"
 )
 
+// EnsureSSHKeypairSecret
 func EnsureSSHKeypairSecret(client kubernetes.Interface) (*rsa.PrivateKey, error) {
+	if client == nil {
+		return nil, fmt.Errorf("got an nil k8s client")
+	}
```
Comment (mrIncompetent) [inline pkg/ssh/configmap.go]: Do we really wan to do a nil check?

File: pkg/ssh/configmap.go [hunk 3]
```diff
@@ -20,48 +20,59 @@ const (
...
-	return keyFromSecret(secret)
 }
 
 func keyFromSecret(secret *v1.Secret) (*rsa.PrivateKey, error) {
 	b, exists := secret.Data[privateKeyDataIndex]
 	if !exists {
 		return nil, fmt.Errorf("key data not found in secret '%s/%s' (secret.data['%s']). remove it and a new one will be created", secret.Namespace, secret.Name, privateKeyDataIndex)
 	}
+	if len(b) == 0 {
+		return nil, fmt.Errorf("key data not found in secret '%s/%s' (secret.data['%s']). remove it and a new one will be created", secret.Namespace, secret.Name, privateKeyDataIndex)
```
Comment (mrIncompetent) [inline pkg/ssh/configmap.go]: i would rather state something like `invalid key data found`

File: pkg/ssh/configmap.go (hunk 2 above)
Comment (guusvw) [reply pkg/ssh/configmap.go]: Yes ;) make the code more stable

File: pkg/ssh/configmap.go (hunk 3 above)
Comment (guusvw) [reply pkg/ssh/configmap.go]: fixed

**Pull request merged**